### PR TITLE
fix(shorebird_cli): `shorebird` should work within child directories

### LINKED ISSUE #1395: fix: shorebird release should support being run from child directories
**Description**

Running `shorebird release ...` from a child directory of the Flutter project's root directory fails with the following error:

```console
Shorebird is not initialized. Did you run "shorebird init"?
```

However, running `flutter build ...` in the same directory produces the following logs before succeeding:

```console
path/to/root/of/project/ios » flutter build ipa
Changing current working directory to: /path/to/root/of/project
```

Because `shorebird release` is a pipe through to `flutter build` as called out in the docs [here](https://docs.shorebird.dev/code_push/release), it should behave similarly and succeed as if it were run in the root of the project.

This was discussed in Discord [here](https://discord.com/channels/1030243211995791380/1125516353370075306/1162224072973877309).

**Steps To Reproduce**

1. Create a flutter project
2. Run `shorebird init`
3. `cd` into a child directory (e.g `ios/` or `android/`)
4. Run `shorebird release ...`

**Expected Behavior**

The command succeeds (assuming no other issues).

### PR DESCRIPTION
**Pull request by felangel**
## Description

- fix(shorebird_cli): `shorebird` should work within child directories (closes #1395)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [X] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by eseidel — APPROVED** [review]
This is very nice.  We could effectively ban Directory.current from our codebase. :)

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 1]
```diff
@@ -201,10 +203,16 @@ Please make sure you are running "shorebird init" from the root of your Flutter
...
+    _addShorebirdYamlToProject(
+      projectRoot: projectRoot,
+      appId: appId,
+      flavors: flavors,
+    );
 
     if (!shorebirdEnv.pubspecContainsShorebirdYaml) {
-      _addShorebirdYamlToPubspecAssets();
+      _addShorebirdYamlToPubspecAssets(
+        shorebirdEnv.getPubspecYamlFile(cwd: projectRoot),
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/init_command.dart:214]: Makes me wonder if shorebirdEnv should have been created with a projectRoot (a later refactor).

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart [hunk 2]
```diff
@@ -278,8 +278,11 @@ ${summary.join('\n')}
     return ExitCode.success.code;
   }
 
-  String get _aotOutputPath =>
-      p.join(Directory.current.path, 'build', 'out.aot');
+  String get _aotOutputPath => p.join(
+        shorebirdEnv.getShorebirdProjectRoot()!.path,
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart:282]: Makes me wonder if there ends up being a ShorebirdEnv which is some sort of at-launch environment and later some sort of ProjectEnv which is resolved and we know has a project directory. 🤷

File: packages/shorebird_cli/lib/src/shorebird_env.dart [hunk 3]
```diff
@@ -94,13 +93,50 @@ class ShorebirdEnv {
   }
 
   /// The `shorebird.yaml` file for this project.
-  File getShorebirdYamlFile() {
-    return File(p.join(Directory.current.path, 'shorebird.yaml'));
+  File getShorebirdYamlFile({Directory? cwd}) {
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/shorebird_env.dart]: I suspect having it optional might end up being a footgun. 🤷

**Review by eseidel — APPROVED** [review]

**Comment by eseidel** [comment]
Please feel encouraged to land as-is, my notes were mostly commentary, not asking for fixes.

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/init_command.dart:214]: Yeah it probably can. I'll have a look in a subsequent PR 👍

File: packages/shorebird_cli/lib/src/shorebird_env.dart (hunk 3 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/shorebird_env.dart]: I refactored to make it required

**Pull request merged**